python/uv/private/lock.bzl

```diff
@@ -40,86 +276,78 @@ def lock(*, name, srcs, out, upgrade = False, universal = True, args = [], **kwa
         name: The name of the target to run for updating the requirements.
         srcs: The srcs to use as inputs.
         out: The output file.
-        upgrade: Tell `uv` to always upgrade the dependencies instead of
-            keeping them as they are.
-        universal: Tell `uv` to generate a universal lock file.
-        args: Extra args to pass to the rule.
-        **kwargs: Extra kwargs passed to the binary rule.
+        args: Extra args to pass to `uv`.
+        env: Passed to `uv`.
+        **kwargs: Extra kwargs passed to the {obj}`py_test` rule.
     """
-    pkg = native.package_name()
-    update_target = name + ".update"
-
-    _args = [
-        "--custom-compile-command='bazel run //{}:{}'".format(pkg, update_target),
-        "--generate-hashes",
-        "--emit-index-url",
-        "--no-strip-extras",
-        "--python=$(PYTHON3)",
-    ] + args + [
-        "$(location {})".format(src)
-        for src in srcs
-    ]
-    if upgrade:
-        _args.append("--upgrade")
-    if universal:
-        _args.append("--universal")
-    _args.append("--output-file=$@")
-    cmd = "$(UV_BIN) pip compile " + " ".join(_args)
-
-    # Make a copy to ensure that we are not modifying the initial list
-    srcs = list(srcs)
-
-    # Check if the output file already exists, if yes, first copy it to the
-    # output file location in order to make `uv` not change the requirements if
-    # we are just running the command.
-    if native.glob([out]):
-        cmd = "cp -v $(location {}) $@; {}".format(out, cmd)
-        srcs.append(out)
-
-    native.genrule(
+    update_target = "{}.update".format(name)
+    locker_target = "{}.run".format(name)
+    maybe_out = _maybe_file(out)
+    out_new = out + ".new"
+    target_compatible_with = kwargs.pop("target_compatible_with", select({
+        "@platforms//os:windows": ["@platforms//:incompatible"],
+        "//conditions:default": [],
+    })) if BZLMOD_ENABLED else ["@platforms//:incompatible"]
```

Comment:
This is hard to read with the select nested inside pop() and the ternary. Please write as a regular if-else.